src/internal-links/handler.js

```diff
@@ -67,64 +39,141 @@ function calculatePriority(links) {
+      JSON.stringify(options),
+    );
+
+    const internal404Links = await rumAPIClient.query(
+      '404-internal-links',
+      options,
+    );
+    const transformedLinks = internal404Links.map((link) => ({
+      urlFrom: link.url_from,
+      urlTo: link.url_to,
+      trafficDomain: link.traffic_domain,
+    }));
+
+    let finalLinks = calculatePriority(transformedLinks);
+
+    finalLinks = finalLinks.filter(async (link) => isLinkInaccessible(link.urlTo, log));
+
+    const auditResult = {
+      brokenInternalLinks: finalLinks,
+      fullAuditRef: auditUrl,
+      finalUrl,
+      auditContext: {
+        interval: INTERVAL,
+      },
+    };
 
-  finalLinks = finalLinks.filter(async (link) => isLinkInaccessible(link.urlTo, log));
+    return {
+      auditResult,
+      fullAuditRef: auditUrl,
+    };
+  } catch (error) {
+    log.error(`[${AUDIT_TYPE}] [Site: ${site.getId()}] audit failed with error: ${error.message}`);
+    return {
+      fullAuditRef: auditUrl,
+      auditResult: {
+        finalUrl: auditUrl,
+        error: `[${AUDIT_TYPE}] [Site: ${site.getId()}] audit failed with error: ${error.message}`,
+        success: false,
+      },
+    };
+  }
+}
 
-  const auditResult = {
-    brokenInternalLinks: finalLinks,
-    fullAuditRef: auditUrl,
+export async function runAuditAndImportTopPagesStep(context) {
+  const { site, log, finalUrl } = context;
+  log.info(`[${AUDIT_TYPE}] [Site: ${site.getId()}] starting audit`);
+  const internalLinksAuditRunnerResult = await internalLinksAuditRunner(
     finalUrl,
-    auditContext: {
-      interval: INTERVAL,
-    },
+    context,
+  );
+
+  return {
+    auditResult: internalLinksAuditRunnerResult.auditResult,
+    fullAuditRef: finalUrl,
+    type: 'top-pages',
+    siteId: site.getId(),
   };
+}
 
+export async function prepareScrapingStep(context) {
+  const {
+    site, log, dataAccess,
+  } = context;
+  const { SiteTopPage } = dataAccess;
+  const topPages = await SiteTopPage.allBySiteIdAndSourceAndGeo(site.getId(), 'ahrefs', 'global');
+
+  log.info(
+    `[${AUDIT_TYPE}] [Site: ${site.getId()}] top pages: ${JSON.stringify(
+      topPages,
+    )}`,
+  );
+  const urls = topPages.map((page) => ({ url: page.getUrl() }));
   return {
-    auditResult,
-    fullAuditRef: auditUrl,
+    urls,
+    siteId: site.getId(),
+    type: 'broken-internal-links',
   };
 }
 
-export async function opportunityAndSuggestions(auditUrl, auditData, context) {
-  const kpiDeltas = calculateKpiDeltasForAudit(auditData);
+export async function opportunityAndSuggestionsStep(context) {
+  const {
+    log, site, finalUrl, audit,
+  } = context;
+  log.info(
+    `[${AUDIT_TYPE}] [Site: ${site.getId()}] latestAuditData`,
+    audit.getAuditResult(),
+  );
+  let { brokenInternalLinks } = audit.getAuditResult();
+  // generate suggestions
+  try {
+    brokenInternalLinks = await generateSuggestionData(
+      finalUrl,
+      audit,
+      context,
+      site,
+    );
+    log.info(
```

Comment:
is this for debug? please remove any log statements that are not needed for production analysis, in order to save log volume on coralogix